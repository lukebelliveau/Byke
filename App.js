import React from 'react';
import { StyleSheet, View } from 'react-native';
import MapView from 'react-native-maps';

const getLocation = (success, error = null, options = {}) => navigator.geolocation.getCurrentPosition(success, error, options);

export default class App extends React.Component {
  constructor() {
    super();
    this.state = {
      region: null,
    };
  };

  componentDidMount() {
    getLocation((location) => {
      this.setState((prevState) => ({
        region: {
          latitude: location.coords.latitude,
          longitude: location.coords.longitude,
          latitudeDelta: 0.0922,
          longitudeDelta: 0.0421,
        }
      }))
    })
  }

  render() {
    return (
      <View style={styles.container}>
        {
          this.state.region === null ? <View /> :
            <MapView
              style={styles.container}
              region={this.state.region}
            />
        }
      </View>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
  },
});
