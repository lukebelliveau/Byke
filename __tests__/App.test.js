import React from 'react';
import App from '../App';

import renderer from 'react-test-renderer';

it('renders without crashing', () => {
  const rendered = renderer.create(<App />).toJSON();
  expect(rendered).toBeTruthy();
});

it('loads the map to user geolocation', () => {
  const app = renderer.create(<App />).toJSON();
  expect(app).toMatchSnapshot();
});


/*
* CONFIG
* */
const mockGetCurrentPosition = (callback) => {callback({
  coords: {
    latitude: 50,
    longitude: 50,
  }
})};

const mockGeolocation = {
  getCurrentPosition: jest.fn(mockGetCurrentPosition),
  watchPosition: jest.fn()
};

global.navigator.geolocation = mockGeolocation;

jest.mock('react-native-maps', () => {
  const React = require.requireActual('react');
  const MapView = require.requireActual('react-native-maps');

  class MockCallout extends React.Component {
    render() {
      return React.createElement('Callout', this.props, this.props.children);
    }
  }

  class MockMarker extends React.Component {
    render() {
      return React.createElement('Marker', this.props, this.props.children);
    }
  }

  class MockMapView extends React.Component {
    render() {
      return React.createElement('MapView', this.props, this.props.children);
    }
  }

  MockCallout.propTypes = MapView.Callout.propTypes;
  MockMarker.propTypes = MapView.Marker.propTypes;
  MockMapView.propTypes = MapView.propTypes;
  MockMapView.Marker = MockMarker;
  MockMapView.Callout = MockCallout;
  return MockMapView;
});