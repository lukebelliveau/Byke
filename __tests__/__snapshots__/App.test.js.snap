// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`loads the map to user geolocation 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <MapView
    region={
      Object {
        "latitude": 50,
        "latitudeDelta": 0.0922,
        "longitude": 50,
        "longitudeDelta": 0.0421,
      }
    }
    style={
      Object {
        "flex": 1,
      }
    }
  />
</View>
`;
